### ISSUE TITLE
Using MIX_ASSET_URL with subfolder not working for CSS url

### ISSUE THREAD

**Issue by gabrielboliveira** [JeffreyWay/laravel-mix#2823]
-   Laravel Mix Version: 6.0.11 (`npm list --depth=0`)
-   Node Version (`node -v`): v14.15.1
-  ~NPM Version (`npm -v`): 6.14.8~ (using Yarn)
-   Yarn Version (`yarn -v`): 1.22.10
-   OS: macOS 11.1

### Description:

Basically, setting `MIX_ASSET_URL` with a subfolder directory is breaking any images imported via CSS/SCSS. Example:

```environment
MIX_ASSET_URL=https://cdn.example.io/some-folder
```

```scss
.icon-categories {
  background-image: url('../../../img/icons/categories.svg');
}
```

The generated CSS tries to load `https://cdn.example.io/images/categories.svg`, ignoring `some-folder`. 

<img width="332" alt="image" src="https://user-images.githubusercontent.com/11093090/106518167-42a80600-64b8-11eb-83ec-aaac61b4250e.png">

In this case, I added [webpack-s3-plugin](https://github.com/MikaAK/s3-plugin-webpack) to upload files to s3, and a subfolder of a bucket, but it may be any other CDNs.

### Steps To Reproduce:

Set `MIX_ASSET_URL` to any path with a subdirectory, as example above. Then import the image on css, and try to load the frontend.

**Comment by DannerTerra** [comment]
Same issue here..

In my case I have a Laravel application running on a subfolder that is different from the main domain.

My .env variables look like this:
APP_URL=http://mydomain.com/subfolder
MIX_ASSET_URL=http://mydomain.com/subfolder

Any update/suggestion?

**Comment by DannerTerra** [comment]
Fixed the issue.  #1026

This option will make mix use the complete url to reference resources in .env:
```
APP_URL=http://mydomain.com/subfolder
ASSET_URL=${APP_URL}
MIX_ASSET_URL=${APP_URL}
```
config/app.php:
`'mix_url' => env('MIX_ASSET_URL', null)`

After adding this configuration, still had issues with url() inside app.css to reference fontawesome files.

Laravel Mix will compile urls as follow, which the browser had issues to find:
`url(/fonts/vendor/@fortawesome/fontawesome-free/webfa-brands-400.eot?0fabb6606be4c45acfeedd115d0caca4)`

So in webpack.mix.js I added:
`mix.setResourceRoot('../');`

This option will compile css fonts with relative paths:
`url(../fonts/vendor/@fortawesome/fontawesome-free/webfa-brands-400.eot?0fabb6606be4c45acfeedd115d0caca4)`

Now it works fine!
